jormungandr/src/blockchain/reference_cache.rs

```diff
@@ -121,11 +121,15 @@ impl RefCacheData {
         }
     }
 
-    fn poll_purge(&mut self) -> Poll<(), timer::Error> {
-        while let Some(entry) = try_ready!(self.expirations.poll()) {
-            self.entries.remove(entry.get_ref());
+    pub fn poll_purge(&mut self) -> Poll<(), timer::Error> {
+        loop {
+            match self.expirations.poll()? {
+                Async::NotReady => return Ok(Async::Ready(())),
```

Comment:
It looks like it does not makes sense for this to be a poll, but with std futures, there will be an explicit context to pass down, so OK.